packer/builder/azure/arm/README.md

```diff
@@ -0,0 +1,115 @@
+# packer-azure-arm
+
+The ARM flavor of packer-azure utilizes the
+[Azure Resource Manager APIs](https://msdn.microsoft.com/en-us/library/azure/dn790568.aspx).
+Please see the
+[overview](https://azure.microsoft.com/en-us/documentation/articles/resource-group-overview/)
+for more information about ARM as well as the benefit of ARM.
+
+## Getting Started
+
+The ARM APIs use OAUTH to authenticate, so you must create a Service
+Principal.  The following articles are a good starting points.
+
+ * [Automating Azure on your CI server using a Service Principal](http://blog.davidebbo.com/2014/12/azure-service-principal.html)
+ * [Authenticating a service principal with Azure Resource Manager](https://azure.microsoft.com/en-us/documentation/articles/resource-group-authenticate-service-principal/)
+
+There are three pieces of configuration you will need as a result of
+creating a Service Principal.
+
+ 1. Client ID (aka Service Principal ID)
+ 1. Client Secret (aka Service Principal generated key)
+ 1. Client Tenant (aka Azure Active Directory tenant that owns the Service Principal)
+
+You will also need the following.
+
+ 1. Subscription ID
+ 1. Resource Group
+ 1. Storage Account
+
+Resource Group is where your storage account is located, and Storage
+Account is where the created packer image will be stored.
+
+The Service Principal has been tested with the following [permissions](https://azure.microsoft.com/en-us/documentation/articles/role-based-access-control-configure/).
+Please review the document for the [built in roles](https://azure.microsoft.com/en-gb/documentation/articles/role-based-access-built-in-roles/)
+for more details.
+
+ * Virtual Machine Contributor
+ * Storage Account Contributor
+ * Network Contributor
+ * Owner
+
+> NOTE: the Owner role is too powerful, and more explicit set of roles is TBD.

```

Comment:
I've logged #183 to track what the set of permissions should be instead of just Owner.